src/api/l_graphics_model.c

```diff
@@ -45,7 +50,12 @@ uint32_t luax_checknodeindex(lua_State* L, int index, Model* model) {
       lovrCheck(nodeIndex != MAP_NIL, "ModelData has no node named '%s'", name);
       return (uint32_t) nodeIndex;
     }
-    case LUA_TNUMBER: return lua_tointeger(L, index) - 1;
+    case LUA_TNUMBER: {
+      uint32_t node = luax_checku32(L, index) - 1;
+      ModelData* data = lovrModelGetInfo(model)->data;
+      lovrCheck(node < data->nodeCount, "Invalid node index '%d'", node + 1);
+      return node;
+    }
```

Comment:
If the bounds checks are getting added here, it probably makes sense to remove bounds checks from the C implementations (`lovrModelAnimate`, `lovrModelGetNodeDrawCount`, etc.).  It probably makes more sense to do the checks in one place in the Lua bindings instead of having to remember to do it in each C function.

Reply:
True, I checked a couple of them and there were some that didn't do any bounds checking in the implementation so I thought lifting it out might be good to unify this. I'll do a pass and remove the checks

Reply:
(In particular, all the node transform functions that call lovrModelGet/SetNodeTransform internally don't do bounds checking currently)